signer/src/proto/convert.rs

```diff
@@ -649,20 +650,34 @@ impl From<proto::DkgBegin> for DkgBegin {
 
 impl From<DkgPrivateBegin> for proto::DkgPrivateBegin {
     fn from(value: DkgPrivateBegin) -> Self {
+        let shares = value
+            .dkg_public_shares
+            .iter()
+            .map(|(i, s)| (*i, proto::SignerDkgPublicShares::from(s.clone())))
+            .collect();
+        let dkg_public_shares = Some(proto::DkgPublicShares { shares });
         proto::DkgPrivateBegin {
             dkg_id: value.dkg_id,
             signer_ids: value.signer_ids,
             key_ids: value.key_ids,
+            dkg_public_shares,
         }
     }
 }
 
 impl From<proto::DkgPrivateBegin> for DkgPrivateBegin {
     fn from(value: proto::DkgPrivateBegin) -> Self {
+        let mut dkg_public_shares = HashMap::new();
+        if let Some(shares) = value.dkg_public_shares {
+            for (id, share) in shares.shares {
+                dkg_public_shares.insert(id, DkgPublicShares::try_from(share).unwrap());
```

Comment:
@djordon I had to unwrap here, since you can't return an error from `impl From`.  We should probably change this and the others to `impl TryFrom`, and deal with the fallout.

Reply:
Ok, `unwrap` calls are now gone